src/libraries/Microsoft.PowerFx.Connectors/Shared/ServiceFunction.cs

```diff
@@ -425,26 +440,59 @@ public bool TryGetParamDefaultValue(string paramName, out Tuple<string, DType> d
 
         public async Task<FormulaValue> InvokeAsync(FormattingInfo context, FormulaValue[] args, CancellationToken cancellationToken)
         {
-            if (_invoker == null)
+            cancellationToken.ThrowIfCancellationRequested();
+
+            FormulaValue result = await (_invoker ?? throw new InvalidOperationException($"Function {Name} can't be invoked.")).InvokeAsync(context, args, cancellationToken).ConfigureAwait(false);
+            ExpressionError er = null;
+
+            if (result is ErrorValue ev && (er = ev.Errors.FirstOrDefault(e => e.Kind == ErrorKind.Network)) != null)
+            {
+                result = FormulaValue.NewError(new ExpressionError() { Kind = er.Kind, Severity = er.Severity, Message = $"{Namespace.ToDottedSyntax()}.{Name} failed: {er.Message}" }, ev.Type);
+            }
+
+            if (result is RecordValue rv && IsPageable && rv.TryGetField(FormulaType.String, _pageLink, out FormulaValue pageLink))
+            {
+                string nextLink = (pageLink as StringValue)?.Value;
+                result = new PagedRecordValue(rv, () => GetNextPage(nextLink, cancellationToken), _maxRows);
+            }
+
+            return result;
+        }
+
+        private FormulaValue GetNextPage(string nextLink, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(nextLink))
             {
-                throw new InvalidOperationException($"Function {Name} can't be invoked.");
+                return FormulaValue.NewBlank();
             }
 
+            FormulaValue result = _invoker.InvokeAsync(nextLink, cancellationToken).ConfigureAwait(false).GetAwaiter().GetResult();
+
+            if (result is RecordValue rv && IsPageable && rv.TryGetField(FormulaType.String, _pageLink, out FormulaValue pageLink))
+            {
+                string nextLink2 = (pageLink as StringValue)?.Value;
+                result = new PagedRecordValue(rv, () => GetNextPage(nextLink2, cancellationToken), _maxRows);
+            }
```

Comment:
this is repeated 3 times, can we simplify?
And the one on 494 is missing a null check